client/src/models/history/CreatePart.ts

```diff
@@ -17,18 +19,31 @@ export const create: Create = async (apiType, params) => {
   return part
 }
 
-export const getBufferGeom = async (productId: number, api: ApiHistory) => {
+// export const getBufferGeom = async (productId: number, api: ApiHistory) => {
+//   if (!api) return
+//   const geoms = await api.createBufferGeometry(productId)
+//   return geoms.map(
+//     geom =>
+//       new THREE.Mesh(
+//         geom,
+//         new THREE.MeshStandardMaterial({ color: new THREE.Color('rgb(52, 89, 87)') }),
+//       ),
+//   )
+// }
+
+export const getScene = async (productId: number, api: ApiHistory) => {
   if (!api) return
-  const geoms = await api.createBufferGeometry(productId)
-  return geoms.map(
-    geom =>
-      new THREE.Mesh(
-        geom,
-        new THREE.MeshStandardMaterial({ color: new THREE.Color('rgb(52, 89, 87)') }),
-      ),
-  )
+  const scene = await api.createScene(productId)
+  scene && colorize(scene)
+  return scene
 }
 
+const colorize = (scene: THREE.Scene) => {
+  const customRed = new Color('rgb(203, 67, 22)')
+  setNodesColor('Part', customRed, scene)
+  setNodesTransparency('Part', 0.5, scene)
```

Comment:
I assume "Part" is the name of the group that will be created in createScene, right? Has it always been like that?

Couldn't this be solved in another way? E.g. could createScene return several things instead of leaving the caller to search for hardcoded names?

For example:
```
createScene: (solids: VID[]) => Promise<{scene: Scene, part: Group, solids: Group[]}>
```
Where "part" is of course already a child of "scene" and the "solids" are also already children of "part".

Reply:
Why not, could be possible, yes.

Reply:
So far, I always thought the scene is somehow "userland" and we just give some hints how it could be done.

Reply:
Since we create the Scene in the function `createScene`, the creation of it is not userland. Therefore the user has to know exactly what names the objects in the scene have to identify them. If he does not know the name he as no change to get for example the "Part".

Reply:
And if he would like to access the solids he has to know about the indexed name as well:
```
mesh.name = 'Solid' + index
```

Reply:
createScene() in solid and history api return now all the created THREE objects, see latest commits in [buerli](https://github.com/awv-informatik/buerli/pull/476). The naming for solids and meshes has been changed. The name of the solids correspond to their id. Meshes do only have a name if they're not merged, then the name corresponds to their graphic id.